src/search.cpp

```diff
@@ -868,61 +868,68 @@ Value Search::Worker::search(
         Piece      captured;
 
         while ((move = mp.next_move()) != Move::none())
-            if (move != excludedMove && pos.legal(move))
-            {
-                assert(pos.capture_stage(move));
+        {
+            assert(move.is_ok());
+
+            if (move == excludedMove)
```

Comment:
join those two if statements, no need for comment

Reply:
```cpp
    if (move == excludedMove || !pos.legal(move))
      continue;
```

Reply:
But in the other move loop, these two if statements are separated, and all the other legality checks have a comment on them, so I feel that this way is more consistent?